app/controllers/events_controller.rb

```diff
@@ -37,4 +37,17 @@ def create_web
     render plain: result.to_json
   end
 
+  def create_with_tree
+
+    originating_data = { url: params[:url] }
+
+    steps = [{ name: 'Apple',  type: 'WebRequest' }]
+
+    workflow = Workflow.build_given_a_hierarchy(first_step: { name: 'Apple', type: 'WebRequest', config: { url: 'http://google.com' }, next_steps: [{ name: 'Orange',  type: 'WebRequest', config: { url: 'http://bing.com' } }, { name: 'Banana',  type: 'WebRequest', config: { url: 'http://yahoo.com' } }] })
```

Comment:
and now we're passing configuration into each event handler.

In this case the configuration is specific to the `type`. the balance of configuration of a type vs a new type is something that will happen based on the types of app we're trying to create. I'm interested to see where types even come from. types made from types sometime, too? and also I'm sure we will string together workflows that can be organized into a single type.

Reply:
Actually, I'm pretty sure that most of our applications are going to be configuration.  We want the event handlers to be very generic, so generic they are applicable to most scenarios.